shopinvader_product_stock/models/shopinvader_variant.py

```diff
@@ -27,8 +27,13 @@ def _get_stock_export_key(self):
             return line.name
 
     def _prepare_stock_data(self):
-        stock_field = self.backend_id.product_stock_field_id.name
-        return {"qty": self[stock_field]}
+        self.ensure_one()
+        stock_field = self.backend_id.product_stock_field_id
+        if stock_field.model == "product.template":
+            qty = self[stock_field.name]
```

Comment:
this does not change anything. You should do `self.shopinvader_product_id[stock_fileld.name]` or `self.product_tmpl_id`